src/core/core.interaction.js

```diff
@@ -293,7 +294,33 @@ module.exports = {
 			if (options.intersect && !intersectsItem) {
 				items = [];
 			}
-			return items;
+
+			// group by dataset
+			var byIndex = items.reduce(function(res, element) {
+				if (!res[element._datasetIndex]) {
```

Comment:
do you mean `=== undefined` ?  this will currently evaluate to true if the array contains the integer 0